File: app/src/main/java/com/xuie/imaginaryandroid/gui/meizhi/MeiZhiAdapter.java
package com.xuie.imaginaryandroid.gui.meizhi;

import android.widget.ImageView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.xuie.imaginaryandroid.R;
import com.xuie.imaginaryandroid.data.BaseBean;
import com.xuie.imaginaryandroid.glide.GlideApp;

import java.util.List;

/**
 * Created by xuie on 17-7-5.
 */

public class MeiZhiAdapter extends BaseQuickAdapter<BaseBean, BaseViewHolder> {
    private static final String TAG = "MeiZhiAdapter";

    public MeiZhiAdapter(List<BaseBean> meiZhis) {
        super(R.layout.item_fragment_meizhi, meiZhis);
    }

    @Override
    protected void convert(BaseViewHolder helper, BaseBean item) {
        helper.setText(R.id.date, item.getDesc());
        GlideApp.with(mContext)
                .load(item.getUrl())
                .placeholder(R.mipmap.meizhi_sample)
                .dontTransform()
                .into((ImageView) helper.getView(R.id.meizhi));
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/gank/meizhi/MeiZhiAdapter.kt
package com.xuie.imaginaryandroid.gui.gank.meizhi

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.xuie.imaginaryandroid.R
import com.xuie.imaginaryandroid.data.BaseBean
import com.xuie.imaginaryandroid.util.GlideUtils

/**
 * Created by xuie on 17-7-5.
 */

class MeiZhiAdapter(meiZhis: List<BaseBean>?) : BaseQuickAdapter<BaseBean, BaseViewHolder>(R.layout.item_fragment_meizhi, meiZhis) {

    override fun convert(helper: BaseViewHolder, item: BaseBean) {
        helper.setText(R.id.date, item.desc)
        GlideUtils.loadImageMeizhi(mContext, item.url, helper.getView(R.id.meizhi))
    }

    companion object {
        private val TAG = "MeiZhiAdapter"
    }
}
